### ISSUE TITLE
Support gcc modules branch

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/compiler-explorer#1284]
https://gcc.gnu.org/wiki/cxx-modules - reported by @grafikrobot

**Comment by mattgodbolt** [comment]
NB once/if we put this on the site there's a real security risk of the BMI feature:

```
|program args... A program to invoke and communicate over its stdin/stdout
```

**Comment by mattgodbolt** [comment]
The compiler's built. We'll need to install and deploy and work around the security issue above before it can go live.

**Comment by partouf** [comment]
It seems modules requires access to the cwd to save modulename.gcm , but for gcc that's the CE root folder which preload denies access to. I would investigate the --help to change the destination path for that -> but it seems beta restricts extra userArguments (on all compilers, not just modules - even -O3 doesn't work)

So for this to work;
* gcc needs to start using the /tmp/ce-etc folder as a cwd
* bugfix the userArguments thing

**Comment by mattgodbolt** [comment]
the user arguments thing should be addressed by 32b4beb978df4cb72c3014f646ffc5c6ec9fce9d

Using the temp directory as a cwd sounds like a win.

**Comment by mattgodbolt** [comment]
And of course the user arguments thing was actually fixed by @partouf in e800deecd3b7b31938e81df6c7a7eb876b2ea42f

**Comment by mattgodbolt** [comment]
This is now live
